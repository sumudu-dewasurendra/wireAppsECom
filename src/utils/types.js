export const BUTTON_TYPES = {
    LABEL: 'label',
    SOLID: 'solid',
}