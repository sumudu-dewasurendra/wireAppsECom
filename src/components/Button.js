import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { BUTTON_TYPES } from '../utils/types';

const Button = ({
    color,
    textColor,
    label,
    type,
    onPress,
    disabled = false
}) => {

    const commonStyle = "justify-center items-center p-2 rounded-lg";

    return (
        <TouchableOpacity
            onPress={onPress}
            disabled={disabled}
            className={`${type === BUTTON_TYPES.LABEL ? 
            commonStyle : 
            (`${commonStyle} bg-${color || 'secondary'}`)
            }`}>
            <Text className={`text-${textColor || 'white'} text-1xl rounded-lg font-bold`}>{label}</Text>
        </TouchableOpacity>
    );
}

export default Button;