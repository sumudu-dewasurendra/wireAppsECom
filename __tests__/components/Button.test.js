import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../src/components/Button';
import { BUTTON_TYPES } from '../../src/utils/types';

test('renders correctly', () => {
    const tree = renderer.create(<Button
        title="Test Button"
        onPress={() => {}}
        type={BUTTON_TYPES.SOLID}
        color={"secondary"}
        textColor={"white"}
        disabled={true}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});