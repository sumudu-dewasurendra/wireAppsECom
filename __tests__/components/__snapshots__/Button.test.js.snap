// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
        },
        {
          "fontWeight": "700",
        },
      ]
    }
  />
</View>
`;
